submissions/HannaSyn/friends-app/index.js

```diff
@@ -0,0 +1,127 @@
+const searchInput = document.querySelector('.search-input');
+const sortAge = document.querySelector('.sort-age');
+const sortName = document.querySelector('.sort-name');
+const mainContent = document.querySelector('.main-content');
+const filterGender = document.querySelector('.gender-filter');
+const ageAscending = document.getElementById('age-ascending');
+const ageDescending = document.getElementById('age-descending');
+const nameAscending = document.getElementById('name-ascending');
+const nameDescending = document.getElementById('name-descending');
+const url = 'https://randomuser.me/api/?results=30';
+let users = [];
+
+const loadUsers = async () => {
+  try {
+    const res = await fetch(url);
+    const { results } = await res.json();
+    users = results;
+  }
+  catch(error) {
+    mainContent.innerHTML = `<p>Oops! Here is the error ${error}</p>`
+  }
+}
+
+function renderCards(arr) {
+  let card = '';
+  
+  arr.forEach(({name, picture, dob, email}) => {
+    const template = `<div class="card">
+      <div class="card__content">
+        <figure class="card__photo">
+          <img src=${picture.large} alt="photo">
+        </figure>
+        <div class="card__info">
+          <p class="card__name">${name.first} ${name.last}</p>
+          <p class="card__age">${dob.age} y.o.</p>
+          <a href="mailto:${email}" class="card__email">${email}</a>
+        </div>
+      </div>
+    </div>`
+    card += template
+  })
+  mainContent.innerHTML = card
+}
+
+function filterBySearch(arr, str) {
+  return arr.filter( el => `${el.name.first}${el.name.last}`.includes(str.toLowerCase()))
+}
+
+function filterByGender(value, arr) {
+  if (value === 'all') {
+    return arr
+  } 
+  return arr.filter(el => el.gender == value)
+}
+
+function findChecked(element) {
+  let checked;
+  let inputs = element.querySelectorAll('input');
+  for (let input of inputs) {
+    if (input.checked) {
```

Comment:
Try this option, if not clear, ask))
```javascript
const searchGender = Array.from(document.querySelectorAll('[name="gender"]'));

function getGenderValue() {
      return searchGender.find(item => item.checked).value;
}
```

Reply:
It's clear, but why my solution is bad?

Reply:
You call this function once and only then to find out the selected gender. It was better to use event delegation for this. But then you have to redo a lot, this is not very suitable.
This option is a bit simple and is another way to use the array method to accomplish this task.
If your option was conceived only for the choice of the chosen floor, then you can simplify it a little.
```javascript
const inputsGender = document.querySelectorAll('[name="gender"]');
function findCheckedGender(element) {
  for (let input of inputsGender) {
    if (input.checked) {
      return input.value;   
    }
  }
}
..........
const gender = findCheckedGender()
```

Reply:
Yep, but what if I will modify this app and add some other filters, I want to reuse this function =)

Reply:
If it's ok, and not a bad practice, I'd like to keep it)
I really appreciate for your review, I get so much new info!